src/fermilib/transforms/_bksf.py

```diff
@@ -0,0 +1,430 @@
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+"""bravyi_kitaev_fast transform on fermionic operators."""
+from __future__ import absolute_import
+from fermilib.ops import InteractionOperator
+import networkx
+import numpy
+from projectq.ops import QubitOperator
+
+
+def bravyi_kitaev_fast(operator):
+    """ Find the Pauli-representation of InteractionOperator for Bravyi-Kitaev
+    Super fast (BKSF) algorithm. Pauli-representation of general
+    FermionOperator is not possible in BKSF. Also, the InteractionOperator
+    given as input must be Hermitian. In future we might provide a
+    transformation for a restricted set of fermion operator.
+
+    Args:
+        operator: Interaction Operator.
+
+    Returns:
+        transformed_operator: An instance of the QubitOperator class.
+
+    """
+    if isinstance(operator, InteractionOperator):
+        return bravyi_kitaev_fast_interaction_op(operator)
+    else:
+        TypeError("operator must be an InteractionOperator.")
+
+
+def bravyi_kitaev_fast_interaction_op(iop, n_qubits=None):
+    """The electronic Hamiltonian is represented in terms of creation and
+    annihilation operators. These creation and annihilation operators could be
+    used to define Majorana modes as follows:
+        c_{2i} = a_i + a^{\dagger}_i,
+        c_{2i+1} = (a_i - a^{\dagger}_{i})/(1j)
+    These Majorana modes can be used to define edge operators B_i and A_{ij}:
+        B_i=c_{2i}c_{2i+1},
+        A_{ij}=c_{2i}c_{2j}
+    using these edge operators the fermionic algebra can be generated and hence
+    all the terms in the electronic Hamiltonian can be expressed in terms of
+    edge operators. The terms in electronic Hamiltonian can be divided into
+    five types (arXiv 1208.5986). We can find the edge operator expression for
+    each of those five types. For example, the excitation operator term in
+    Hamiltonian when represented in terms of edge operators becomes:
+    a_i^{\dagger}a_j+a_j^{\dagger}a_i = (-1j/2)*(A_ij*B_i+B_j*A_ij)
+    For the sake of brevity the reader is encouraged to look up the expressions
```

Comment:
@babbush:
Should we put all terms here? People looking up the documentation online don't see the code...
On the other hand, there is always a risk that this documentation might not correspond to the exact code, so looking up the code is a safer things to do...